lib/formats/bs_format_ripejson.c

```diff
@@ -220,6 +226,46 @@ static int process_common_fields(bgpstream_format_t *format,
   return 0;
 }
 
+// readline
+static int readline(bgpstream_format_t *format)
+{
+  int total_read = 0;
+  int current_read = 0;
+  char *buffer_ptr = STATE->json_string_buffer;
+
+  while(1){
+    // read line, including linebreak
+    current_read = bgpstream_transport_readline(
+      format->transport, buffer_ptr, BUFSIZE - total_read, 0);
+    total_read+=current_read;
+
+    if(current_read == 0){
+      // reach EOF, no bytes read, return 0
+      return 0;
+    }
+
+    if(current_read < BUFSIZE - total_read -1 ||
+      buffer_ptr[current_read-1] == '\n'){
+      // if read less than buffer, or last byte is an newline, readline is finished
+
+      // replace linebreak with null
+      buffer_ptr[current_read-1] = '\0';
```

Comment:
what if there was no newline in the input data?
e.g., we reached EOF (so the first half of the if is true), but the last character was not a newline.
i think in this case you're clobbering a meaningful character?

Reply:
Correct. Working on a fix now.

Reply:
fixed in e0f2603c68b265df27ba39014439d753f8ceaeb7